homeassistant/components/duckdns/__init__.py

```diff
@@ -42,14 +42,22 @@
     token = config[DOMAIN][CONF_ACCESS_TOKEN]
     session = async_get_clientsession(hass)
 
-    result = await _update_duckdns(session, domain, token)
+    await _update_duckdns(session, domain, token)
 
-    if not result:
-        return False
+    fail = 0
+    enable = 0
 
     async def update_domain_interval(now):
         """Update the DuckDNS entry."""
-        await _update_duckdns(session, domain, token)
+        nonlocal fail, enable
+        if enable >= 0:
+            if await _update_duckdns(session, domain, token):
+                fail = 0
+            else:
+                fail += 1
+                enable = -11 if fail > 2 else -2
```

Comment:
These numbers are not clearly connected to the tone interval which makes it hard to follow the logic.

Can we change the logic to clearly use and modify time intervals and rescheduling of the next function call?

Reply:
I agree, there are better code example in our code base how we can provide a retry. That one looks a bit hacky